### PR TITLE
simplify logic in cnvkit batch + fix pon tests

### PR DESCRIPTION
**Pull request by maxulysse**
replace #2556 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by lassefolkersen — APPROVED** [review]
I think it looks good. I actually completely forgot why it was all wrapped in if statements for the different inputs.

**Comment by lassefolkersen** [comment]
I mean your tests do fail 
```
FAILED tests/test_versions_yml.py::test_ensure_valid_version_yml[cnvkit batch test_cnvkit_cram_wgs_somatic] - AssertionError: END_VERSIONS detected in versions.yml. This is a sign of an ill-formatted HEREDOC
```
So maybe I was a bit fast with that 'approve' 😄

**Comment by maxulysse** [comment]
> I mean your tests do fail
> 
> ```
> FAILED tests/test_versions_yml.py::test_ensure_valid_version_yml[cnvkit batch test_cnvkit_cram_wgs_somatic] - AssertionError: END_VERSIONS detected in versions.yml. This is a sign of an ill-formatted HEREDOC
> ```
> 
> So maybe I was a bit fast with that 'approve' smile

true, but no idea why they're failing...
At least the logic is now working, and it is simplified

File: modules/nf-core/cnvkit/batch/main.nf [hunk 1]
```diff
@@ -63,133 +64,39 @@ process CNVKIT_BATCH {
...
+        normal_args = "--normal $pon_input"
     }
 
+    def target_args = targets ? "--targets $targets" : ""
+    def reference_args = reference ? "--reference $reference" : ""
+
+    def samtools_cram_convert = ''
+    samtools_cram_convert += normal_cram ? "samtools view -T $fasta $fai_reference $normal -@ $task.cpus -o $normal_out\n" : ''
+    samtools_cram_convert += tumor_cram ? "samtools view -T $fasta $fai_reference $tumor -@ $task.cpus -o $tumor_out\n" : ''
+    def samtools_version = normal_cram || tumor_cram ? "samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')" : ''
```
Comment (SusiJo) [inline modules/nf-core/cnvkit/batch/main.nf:81]: ❤️  the `samtools_cram_convert` block - very elegant!

File: modules/nf-core/cnvkit/batch/main.nf [hunk 2]
```diff
@@ -63,133 +64,39 @@ process CNVKIT_BATCH {
...
+
+    """
+    $samtools_cram_convert
+
+    cnvkit.py \\
+        batch \\
+        $tumor_out \\
+        $normal_args \\
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/nf-core/cnvkit/batch/main.nf:89]: This is all done by the underlying logic that set up the multiple variables

**Review by SusiJo — APPROVED** [review]
Thanks for simplifying! Looks amazing! 🚀 

Hope it's all working in germline/tumoronly mode. I remember that the usage of the command-line flags in cnvkit was a bit confusing.  

Please make sure the `software_versions.yml` is working fine before merging :)

File: modules/nf-core/cnvkit/batch/main.nf (hunk 1 above)
Comment (marrip) [inline modules/nf-core/cnvkit/batch/main.nf:81]: I am suspecting that this generates an empty line in the `versions.yml` whenever `samtools` is not used. You could adapt for that adding a line break and a tab and putting everything in one line. Maybe not the most elegant...  
```suggestion
    def samtools_version = normal_cram || tumor_cram ? "samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$/\n\t/')" : ''
```

File: modules/nf-core/cnvkit/batch/main.nf [hunk 3]
```diff
@@ -63,133 +64,39 @@ process CNVKIT_BATCH {
...
+        $fasta_args \\
+        $reference_args \\
+        $target_args \\
+        --processes $task.cpus \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        $samtools_version
+        cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
```
Comment (marrip) [inline modules/nf-core/cnvkit/batch/main.nf:99]:
```suggestion
        ${samtools_version}cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
```

**Review by marrip — COMMENTED** [review]
Suggestion to fix `versions.yml`

**Comment by marrip** [comment]
👏

**Pull request merged**